api/src/bigquerytest/java/org/pmiops/workbench/api/DataSetControllerBQTest.java

```diff
@@ -0,0 +1,504 @@
+                WORKSPACE_NAME,
+                KernelTypeEnum.PYTHON.toString(),
+                createDataSetRequest(
+                    Arrays.asList(dbConditionConceptSet),
+                    Arrays.asList(dbCohort1),
+                    Arrays.asList(Domain.CONDITION),
+                    false,
+                    PrePackagedConceptSetEnum.NONE))
+            .getBody()
+            .getCode();
+    assertThat(code)
+        .contains(
+            "import pandas\n"
+                + "import os\n"
+                + "\n"
+                + "# This query represents dataset \"null\" for domain \"condition\"\n"
+                + "dataset_00000000_condition_sql =");
+
+    String query = extractPythonQuery(code, 1);
+
+    try {
+      TableResult result =
+          bigQueryService.executeQuery(
+              QueryJobConfiguration.newBuilder(query).setUseLegacySql(false).build());
+      assertThat(result.getTotalRows()).isEqualTo(1L);
+    } catch (Exception e) {
+      fail("Problem generating BigQuery query for notebooks: " + e.getCause().getMessage());
+    }
+  }
+
+  @Test
+  public void testGenerateCodeR() {
+    String code =
+        controller
+            .generateCode(
+                WORKSPACE_NAMESPACE,
+                WORKSPACE_NAME,
+                KernelTypeEnum.R.toString(),
+                createDataSetRequest(
+                    Arrays.asList(dbConditionConceptSet),
+                    Arrays.asList(dbCohort1),
+                    Arrays.asList(Domain.CONDITION),
+                    false,
+                    PrePackagedConceptSetEnum.NONE))
+            .getBody()
+            .getCode();
+
+    assertThat(code)
+        .contains(
+            "library(bigrquery)\n"
+                + "\n# This query represents dataset \"null\" for domain \"condition\"\n"
+                + "dataset_00000000_condition_sql <- paste(\"");
+    String query = extractRQuery(code);
+
+    try {
+      TableResult result =
+          bigQueryService.executeQuery(
+              QueryJobConfiguration.newBuilder(query).setUseLegacySql(false).build());
+      assertThat(result.getTotalRows()).isEqualTo(1L);
+    } catch (Exception e) {
+      fail("Problem generating BigQuery query for notebooks: " + e.getCause().getMessage());
+    }
+  }
+
+  @Test
+  public void testGenerateCodeTwoConceptSets() {
+    String code =
+        controller
+            .generateCode(
+                WORKSPACE_NAMESPACE,
+                WORKSPACE_NAME,
+                KernelTypeEnum.PYTHON.toString(),
+                createDataSetRequest(
+                    Arrays.asList(dbConditionConceptSet, dbProcedureConceptSet),
+                    Arrays.asList(dbCohort1),
+                    Arrays.asList(Domain.CONDITION, Domain.PROCEDURE),
+                    false,
+                    PrePackagedConceptSetEnum.NONE))
+            .getBody()
+            .getCode();
+
+    String query = extractPythonQuery(code, 1);
```

Comment:
should we check for assert for Comments like in previous test

Reply:
Thanks for catching this. I extracted a private method that each test calls to capture the same asserts for all.